### ISSUE TITLE
Unable to create ASP.NET 5 project template

### ISSUE THREAD

**Issue by msmolcic** [ligershark/side-waffle#312]
I'm trying to create ASP.NET 5 project template, but when I choose my project inside Add Template Reference (SideWaffle project) and click OK I get the following error:

> "Did not find new element in the hierarchy for item "_project1.vstemplate". The item would have been added with type "Content". The project item schema service knows about this item type. The item does not exist in project file."

Application's .csproj file is not added to VSIX and I can't rebuild solution successfully as I get:

> The project file coud not be loaded. Could not find file '...VSIXTest\WebApplicationTest\WebApplication.csproj'

**Comment by sayedihashimi** [comment]
@msmolcic did you see https://github.com/ligershark/side-waffle/wiki/How-to-create-templates-for-ASP.NET-5?

Also check the value in `_project.template.xml` for ```<Project File="AspNetProject.xproj" ReplaceParameters="true">``` to make sure the `File` attribute has the correct value and it should *not* have a `TargetFileName` attribute.

**Comment by msmolcic** [comment]
> Also check the value in _project.template.xml for <Project File="AspNetProject.xproj" ReplaceParameters="true"> to make sure the File attribute has the correct value and it should not have a TargetFileName attribute.

This worked out for me. Thanks a lot!

**Comment by sayedihashimi** [comment]
Ok glad you got it working

**Comment by MisterGoodcat** [comment]
This is not working for me. I've tried several variations and different versions (e.g. template builder stable/pre-releases etc.) and always get the same result. Basically:

* Create a VSIX project
* Add a ASP.NET 5 project (e.g. empty template)
* Right-click VSIX and add template reference
* The _project.vstemplate.xml and _preprocess.xml in the target project are created correctly
* However I immediately receive the error above: 

"Did not find new element in the hierarchy for item "_project1.vstemplate". The item would have been added with type "Content". The project item schema service knows about this item type. The item does not exist in the project file."

I tried to simply ignore that and continue with editing the _project.vstemplate.xml/_preprocess.xml as advised etc., but then when I want to use the template in VS I get:

"The specified default content type (Default) is not defined in the project item schema."

... and no project is actually created.

VS is 2015 Enterprise, ASP.NET is Beta 8.

Is this a known issue? Any hints what I'm doing wrong?

**Comment by MisterGoodcat** [comment]
I figured it out. The second error was caused by the experimental instance of VS running with a different language pack. Once fixed, the template could be used correctly.

The first error message however is still relevant, maybe the instructions should be updated to reflect that.

**Comment by sayedihashimi** [comment]
Can you share the code so we can take a look?

**Comment by MisterGoodcat** [comment]
The second error seems only dependent on the installed/configured environment. In particular, I have a German version of Visual Studio 2015 and installed the English language pack on top of it. My (English) development instance fired up the experimental instance in German, and that seems to have caused the issue. Once I switched the experimental instance to English, the error was gone.

The first error pretty much only is:
* File/New project/VSIX
* File/New project/ASP.NET 5 (I used the empty template)
* Right-click VSIX project and add template reference => error message

After the error the .xproj entry in the VSIX does not show up (the XML files in the ASP.NET 5 projects are there); however, if I close the solution and re-open it, the .xproj is there and I can continue from that. If you really need that code (it's really only these two empty projects), I can share it.
